include/world_builder/bounding_box.h

```diff
@@ -0,0 +1,446 @@
+   * space in which an object is contained; if a candidate point is not within
+   * the bounding box (a test that is cheap to execute), then it is not necessary
+   * to perform an expensive test whether the candidate point is in fact inside
+   * the object itself. Bounding boxes are therefore often used as a first,
+   * cheap rejection test before more detailed checks. As such, bounding boxes
+   * serve many of the same purposes as the
+   * [convex hull](https://en.wikipedia.org/wiki/Convex_hull), for which it is
+   * also relatively straightforward to compute whether a point is inside or
+   * outside, though not quite as cheap as for the bounding box.
+   *
+   * Taking the cross section of a BoundingBox<spacedim> orthogonal to a given
+   * direction gives a box in one dimension lower: BoundingBox<spacedim - 1>.
+   * In 3D, the 2 coordinates of the cross section of BoundingBox<3> can be
+   * ordered in 2 different ways. That is, if we take the cross section orthogonal
+   * to the y direction we could either order a 3D-coordinate into a
+   * 2D-coordinate as $(x,z)$ or as $(z,x)$. This class uses the second
+   * convention, corresponding to the coordinates being ordered cyclicly
+   * $x \rightarrow y \rightarrow z \rightarrow x \rightarrow ... $
+   * To be precise, if we take a cross section:
+   *
+   * | Orthogonal to | Cross section coordinates ordered as |
+   * |:-------------:|:------------------------------------:|
+   * |      x        |               (y, z)                 |
+   * |      y        |               (z, x)                 |
+   * |      z        |               (x, y)                 |
+   *
+   * This is according to the convention set by the function
+   * <code>coordinate_to_one_dim_higher</code>.
+   *
+   * @note The majority of this class is copied from the deal.II library.
+   */
+  template <int spacedim>
+  class BoundingBox
+  {
+    public:
+      /**
+       * Standard constructor. Creates an object that corresponds to an empty box,
+       * i.e. a degenerate box with both points being the origin.
+       */
+      BoundingBox() = default;
+
+      /**
+       * Standard constructor for non-empty boxes: it uses a pair of points
+       * which describe the box: one for the bottom and one for the top
+       * corner.
+       */
+      BoundingBox(const std::pair<Point<spacedim>, Point<spacedim>>
+                  &boundary_points);
+
+      /**
+       * Construct the bounding box that encloses all the points in the given
+       * container.
+       *
+       * The constructor supports any Container that provides begin() and end()
+       * iterators to Point<spacedim> elements.
+       */
+      template <class Container>
+      BoundingBox(const Container &points);
+
+      /**
+       * Return a reference to the boundary_points
+       */
+      std::pair<Point<spacedim>, Point<spacedim>> &
+                                               get_boundary_points();
```

Comment:
This alignment is freaking me of a bit. Could you either get it on one line or alligned with the beginning of the line. apparently astyle doesn't fix this.